### DISCUSSION TITLE
Implement In Memory File System Provider

### DISCUSSION THREAD

**Discussion by ZCH2005** [TypeFox/monaco-languageclient#658] [Q&A] [answered]
Hi @kaisalmen,

We are using Monaco Editor Wrapper and Langium language server. From the Monaco Editor wrapper, we start the Langium language server by web worker.  Now we want to add an in-memory file system as shown in [builtin-libraries](https://langium.org/guides/builtin-library/), in the last part, an DslLibraryFileSystemProvider was implemented in client side (vs code extension). In Monaco Editor Wrapper, how to implement the similar DslLibraryFileSystemProvider?

**Comment by kaisalmen** [comment, marked as answer]
Hi @ZCH2005 you should be able to do very similarly. 
`@codingame/monaco-vscode-files-service-override` is always configured and you can use the [registerCustomProvider](https://github.com/CodinGame/monaco-vscode-api/blob/main/src/service-override/files.ts#L585) to register you own file system provider. 

The other possibility is to do it like [this](https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/src/python/client/main.ts#L21-L30) and use the readily available `RegisteredFileSystemProvider`.
